lib/Service/Appointments/EventConflictFilter.php

```diff
@@ -56,14 +60,31 @@ public function filter(AppointmentConfig $config, array $slots): array {
 		}
 		// Always check the target calendar for conflicts
 		$query->addSearchCalendar($config->getTargetCalendarUri());
+		$query->addType('VEVENT');
 		$preparationDuration = DateInterval::createFromDateString($config->getPreparationDuration() . ' seconds');
 		$followUpDuration = DateInterval::createFromDateString($config->getFollowupDuration() . ' seconds');
-
+		$this->logger->debug('Appointment config ' . $config->getToken() . ' is looking with preparation duration {preparation} and follow up duration {followup} in calendar {calendarUri}', [
+			'preparation' => $preparationDuration->s,
+			'followup' => $followUpDuration->s,
+			'calendarUri' => $config->getTargetCalendarUri()
+		]);
 		return array_filter($slots, function (Interval $slot) use ($followUpDuration, $preparationDuration, $query): bool {
 			$query->setTimerangeStart($slot->getStartAsObject()->sub($preparationDuration));
 			$query->setTimerangeEnd($slot->getEndAsObject()->add($followUpDuration));
 
-			$objects = $this->calendarManager->searchForPrincipal($query);
+			$events = $this->calendarManager->searchForPrincipal($query);
+			$objects = array_filter($events, function (array $event) {
+				$isCancelled = ($event['objects'][0]['STATUS'][0] ?? null) === 'CANCELLED';
+				$isFree = ($event['objects'][0]['TRANSP'][0] ?? null) === 'TRANSPARENT';
+				if (!$isCancelled && !$isFree) {
+					$this->logger->debug('Appointment config conflicting with event with UID {uid}', [
```

Comment:
could you move the log outside of this `array_filter`, log all UIDs at once (implode with `:` or similar) and include the time stamp of `$slot->getStartAsObject()->sub($preparationDuration)` and `$slot->getEndAsObject()->add($followUpDuration)`? Those timestamps will tell more than the preparation/followup second above because we'll know the precise window that is searched :)

Reply:
Absolutely. I meant all *conflicting* UIDs :)